E0079 needs to be updated to new format

From: src/test/compile-fail/E0079.rs

Error E0079 needs a span_label, updating it from: 

```
error[E0079]: mismatched types
  --> src/test/compile-fail/E0079.rs:12:9
   |
12 |     Q = "32" //~ ERROR E0079
   |         ^^^^
   |
   = note: expected type `isize`
   = note:    found type `string literal`
```

To:

```
error[E0079]: mismatched types
  --> src/test/compile-fail/E0079.rs:12:9
   |
12 |     Q = "32" //~ ERROR E0079
   |         ^^^^ expected `isize` type
   |
   = note: expected type `isize`
   = note:    found type `string literal`
```

--- a/src/librustc_typeck/collect.rs
+++ b/src/librustc_typeck/collect.rs
@@ -1057,6 +1057,7 @@ fn convert_struct_def<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
         let print_err = |cv: ConstVal| {
             struct_span_err!(ccx.tcx.sess, e.span, E0079, "mismatched types")
                 .note_expected_found(&"type", &ty_hint, &format!("{}", cv.description()))
+                .span_label(e.span, &format!("expected '{}' type", ty_hint))
                 .emit();
         };
 

--- a/src/test/compile-fail/E0079.rs
+++ b/src/test/compile-fail/E0079.rs
@@ -10,6 +10,7 @@
 
 enum Foo {
     Q = "32" //~ ERROR E0079
+    //~^ expected 'isize' type
 }
 
 fn main() {

